docs/extensions/referenced-resources.md

```diff
@@ -0,0 +1,29 @@
+# Referenced Resources
```

Comment:
Nit: Can you add the document to the index, https://github.com/gardener/gardener/blob/master/docs/README.md#extensions?

Reply:
Done